sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/regexpExpressions.scala

```diff
@@ -54,9 +54,20 @@ trait StringRegexExpression extends Expression
     Pattern.compile(escape(str))
   }
 
+  var lastPatternStr: String = null
```

Comment:
You can keep `lastPattern` as `Pattern`, and compare it to the current pattern via `lastPattern.pattern() == patternStr`. No need to keep `lastPattern` as a string.

Reply:
Of course，I can do like this. I just followed the way of `RegExpExtract` and `RegExpReplace` here.